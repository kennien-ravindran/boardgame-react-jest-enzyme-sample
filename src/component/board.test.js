import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Board from './board';
import TicTacToeModel from '../model/tictactoeModel';

Enzyme.configure({ adapter: new Adapter() })
const boardData = TicTacToeModel.generateBoardData(3);
const updateTileCallbackFn = {}
const gameOverCallbackFn = {}
describe('Board', () => {
  it('should render correctly', () => {
    const output = shallow(
      //<Board boardData={boardData}/>
      <Board boardData={boardData} updateTileCallback={updateTileCallbackFn} gameOverCallback={gameOverCallbackFn}/>

    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});
