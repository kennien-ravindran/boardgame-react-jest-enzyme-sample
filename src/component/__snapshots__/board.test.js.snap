// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render correctly 1`] = `
<table>
  <tbody>
    <tr
      key="0"
    >
      <td
        key="0x0"
      >
        <Tile
          column={0}
          onClick={[Function]}
          row={0}
          value=""
        />
      </td>
      <td
        key="0x1"
      >
        <Tile
          column={1}
          onClick={[Function]}
          row={0}
          value=""
        />
      </td>
      <td
        key="0x2"
      >
        <Tile
          column={2}
          onClick={[Function]}
          row={0}
          value=""
        />
      </td>
    </tr>
    <tr
      key="1"
    >
      <td
        key="1x0"
      >
        <Tile
          column={0}
          onClick={[Function]}
          row={1}
          value=""
        />
      </td>
      <td
        key="1x1"
      >
        <Tile
          column={1}
          onClick={[Function]}
          row={1}
          value=""
        />
      </td>
      <td
        key="1x2"
      >
        <Tile
          column={2}
          onClick={[Function]}
          row={1}
          value=""
        />
      </td>
    </tr>
    <tr
      key="2"
    >
      <td
        key="2x0"
      >
        <Tile
          column={0}
          onClick={[Function]}
          row={2}
          value=""
        />
      </td>
      <td
        key="2x1"
      >
        <Tile
          column={1}
          onClick={[Function]}
          row={2}
          value=""
        />
      </td>
      <td
        key="2x2"
      >
        <Tile
          column={2}
          onClick={[Function]}
          row={2}
          value=""
        />
      </td>
    </tr>
  </tbody>
</table>
`;
